src/judging_round_display.js

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+
+
+class JudgingRoundDisplay extends React.Component {
+    state = {
+        data: {},
+    }
+
+    componentDidUpdate(prevProps) {
+	const judging_round_id = this.props.judging_round
+	if (judging_round_id &&
+	      prevProps.judging_round !== this.props.judging_round) {
+	    this.fetch_data()
+	}
+    }
+
+    fetch_data() {
+	fetch("http://localhost:8000/api/v1/judging_round/" +
+	      judging_round_id + "/",
+	      {credentials: "include", mode: "cors"})
+	    .then(res => res.json()).then(data => {
+		this.setState({ data: data })
+	    })
+    }
+
+    
+    render() {
+	if (this.props.judging_round) {
+	    return (<ul>
+		    <li>Props Judging Round: {this.props.judging_round}</li>
+		    <li>Full Name: {this.state.data.full_name}</li>
+		    <li>Id: {this.state.data.id}</li>
```

Comment:
There's a lot of drilling into `this.state.data` throughout this render and you could pull some constants out using destructing above. Something like:

```js
render() {
  if (this.props.judging_round) {
    const { judging_round } = this.props
    const { full_name, id, is_active, round_type, cycle_based_round, program_id } = this.state.data
    return (
      <ul>
        <li>Props Judging Round: {judging_round}</li>
...etc...
    )
  } else {
    return (<div>No Judging Round Selected</div>)
  }
}
```

Reply:
Didn't go quite as far as you suggest, but at least removed some of the extra drilling.